### PR TITLE
Adding course run creation into the course endpoint so they can be wr…

### PR DESCRIPTION
**Pull request by Dillon-Dumesnil**
…apped in an atomic transaction

### CODE REVIEW

File: course_discovery/apps/api/v1/views/courses.py [hunk 1]
```diff
@@ -194,6 +196,13 @@ def create(self, request, *args, **kwargs):
             draft=True,
         )
 
+        # We want to create the course run here so it is captured as part of the atomic transaction
+        # and we can roll all of these changes back if the course run creation fails.
```
Comment (staubina) [inline course_discovery/apps/api/v1/views/courses.py]: Nit: I think you could remove the part of the comment `and we can roll all of these changes back if the course run creation fails.`, AtomicTransaction implies it.  Not a blocker.

File: course_discovery/apps/api/v1/views/courses.py [hunk 2]
```diff
@@ -194,6 +196,13 @@ def create(self, request, *args, **kwargs):
...
+        # Note: We have to send the request object as well because it is used for its metadata
+        # (like request.user and is set as part of the serializer context)
+        if course_run_creation_fields:
+            CourseRunViewSet().create_run_helper(course_run_creation_fields, request)
```
Comment (staubina) [inline course_discovery/apps/api/v1/views/courses.py:204]: I misunderstood your description, this looks fine to me.

Comment (staubina) [reply course_discovery/apps/api/v1/views/courses.py:204]: I like that it follows the same path as the course run endpoint

👍

**Review by staubina — CHANGES_REQUESTED** [review]
The changes seem straight forward, but I think we should add/update some testing as it will be a critical issue if this workflow is altered in a bad way.

File: course_discovery/apps/api/v1/views/courses.py (hunk 2 above)
Comment (Dillon-Dumesnil) [reply course_discovery/apps/api/v1/views/courses.py:204]: That was the goal! Just a bit annoying making it work.

**Review by staubina — APPROVED** [review]

**Pull request merged**